### PR TITLE
Fix Jackson Conflict + Elasticsearch Support Bump

### PR DESCRIPTION
**Pull request by szareiangm**
I fixed the issue caused after my previous contribution and also upgraded support for the new Elasticsearch.

CC: @nafid

### CODE REVIEW

File: elasticsearch/src/test/scala/com/snowplowanalytics/stream/loader/clients/ElasticsearchBulkSenderSpec.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ import org.specs2.mutable.Specification
 class ElasticsearchBulkSenderSpec extends Specification {
   val node = LocalNode("es", System.getProperty("java.io.tmpdir"))
   node.start()
-  val client       = node.elastic4sclient()
+  val client       = node.client(true)
```
Comment (szareiangm) [inline elasticsearch/src/test/scala/com/snowplowanalytics/stream/loader/clients/ElasticsearchBulkSenderSpec.scala]: there is no `elastic4sclient` anymore and that `true` is for `shutdown after close`

File: project/Dependencies.scala [hunk 2]
```diff
@@ -21,41 +21,52 @@ object Dependencies {
...
     val validator        = "2.2.6"
-    val elasticsearch    = "6.2.4"
+    val elasticsearch    = "6.3.2"
     val nsqClient        = "1.1.0-rc1"
+    val jackson          = "2.9.6"
+
+    // TODO: consider removing paranamer version override when upgrading other libaries
+    val paranamer = "2.8" // Jackson 2.9.6 uses version 2.8 of paranamer internally
+                          // but there is json4s-core_2.12 3.2.11 that overrides it back to 2.6
+
```
Comment (szareiangm) [inline project/Dependencies.scala]: `paranamer` was source of the Jackson issue. Jackson is used internally in `elastic4s` and not changable and Jackson uses `paranamer` for detecting fields and methods of a pojo. There is `json4s` dependency that I guess comes form snowplow analytics sdk. that one has older `paranamer` and caused the issue.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -21,41 +21,52 @@ object Dependencies {
...
+    val config           = "com.typesafe"                     %  "config"                       % V.config
+    val slf4j            = "org.slf4j"                        %  "slf4j-simple"                 % V.slf4j
+    val log4jOverSlf4j   = "org.slf4j"                        %  "log4j-over-slf4j"             % V.slf4j
+    val kinesisClient    = "com.amazonaws"                    %  "amazon-kinesis-client"        % V.kinesisClient
+    val kinesisConnector = "com.amazonaws"                    %  "amazon-kinesis-connectors"    % V.kinesisConnector
+    val validator        = "com.github.fge"                   %  "json-schema-validator"        % V.validator
+    val elasticsearch    = "org.elasticsearch"                %  "elasticsearch"                % V.elasticsearch
+    val nsqClient        = "com.snowplowanalytics"            %  "nsq-java-client_2.10"         % V.nsqClient
+    val paranamer        = "com.thoughtworks.paranamer"       %  "paranamer"                    % V.paranamer
+    val jacksonCbor      = "com.fasterxml.jackson.dataformat" %  "jackson-dataformat-cbor"      % V.jackson
```
Comment (szareiangm) [inline project/Dependencies.scala]: due to `com.fasterxml.jackson.dataformat` length I had to add space to keep the structure.

File: build.sbt [hunk 4]
```diff
@@ -19,6 +19,9 @@ lazy val commonDependencies = Seq(
   Dependencies.Libraries.kinesisConnector,
   Dependencies.Libraries.validator,
   Dependencies.Libraries.nsqClient,
+  Dependencies.Libraries.paranamer,
+  Dependencies.Libraries.jacksonDatabind,
+  Dependencies.Libraries.jacksonCbor,
```
Comment (szareiangm) [inline build.sbt]: `jacksonDatabind` and `jacksonCbor` solved the issue with Kinesis client exception reported by the user and `paranamer` solved the `elastic4s` side.

File: build.sbt [hunk 5]
```diff
@@ -15,10 +15,15 @@ lazy val commonDependencies = Seq(
   Dependencies.Libraries.config,
   Dependencies.Libraries.slf4j,
   Dependencies.Libraries.log4jOverSlf4j,
+  Dependencies.Libraries.log4jApi,
+  Dependencies.Libraries.log4jCore,
```
Comment (szareiangm) [inline build.sbt]: I had to add this to fix the issue with previous build failure: https://travis-ci.org/snowplow/snowplow-elasticsearch-loader/jobs/420217876

**Comment by snowplowcla** [comment]
@szareiangm  has signed the Software Grant and Corporate Contributor License Agreement

**Pull request merged**